taskcluster/ci/push-addon/kind.yml

```diff
@@ -0,0 +1,45 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - mozillavpn_taskgraph.transforms.build:transforms # So we can do "requires-level"
+    - mozillavpn_taskgraph.transforms.release_index:transforms
+    - taskgraph.transforms.job:transforms
+    - taskgraph.transforms.task:transforms
+
+kind-dependencies:
+    - build
+
+jobs:
+  stage/deploy:
+      description: "Addons Stage Deploy"
+      treeherder:
+          symbol: D
+          kind: build
+          tier: 1
+          platform: addons/all
+      worker-type: b-linux
+      requires-level: 3
+      add-index-routes: addons # Name of the index
+      scopes:
+         - 'secrets:get:project/mozillavpn/tokens'
+      dependencies:
+        build: "build-addons"
+      fetches:
+        build:
+            - artifact: addons.zip
+      worker:
+          docker-image: {in-tree: wasm}
+          max-run-time: 3600
+          artifacts:
+              - type: directory
+                name: public/build
+                path: /builds/worker/artifacts
+      run:
+          using: run-task
+          use-caches: true
+          cwd: '{checkout}'
+          command: ./taskcluster/scripts/push-addons/deploy.sh
```

Comment:
Is the intent of this kind to push addons to archive.mozilla.org?

If so unfortunately you'll have to redo this task to use our "beetmover" worker (beetmover -> bit mover -> move bits to archive.mozilla.org.. don't ask). This is the standard and secure method we have in place for moving things over to archive.mozilla.org, so I think it'll be a requirement.

The good news is I've already requested and got access to the credentials we can use to move things under `pub/vpn/addons`, and these credentials are already in the right place. @gabrielBusta is looking into standing up some tasks that will be able to move the main client there. So if you block on his work, a lot of the scaffolding will be done for you already.

Alternatively, I can put you on the right path if you want to go ahead with it yourself.

Reply:
Happy to dig into this, once we have the beetmover stuff setup :) 

I do think the "end goal" is exactly this. For now, as we are testing - we're fine with this task just unpacking the bundle and using the release-index work to get a permalink to each file. (I kind of to have this even without archive, so we can stop using gh-pages as our current "file" storage for vpn :D )

Reply:
Cool, sounds good!

Reply:
@strseb @ahal reviving this thread. We can now beetmove TC artifacts to `pub/vpn/addons`, is the `addons.zip` bundle what we want to push to the archive? And how should we trigger this beetmoving? a Taskcluster hook?

Reply:
Yeah, I was thinking there'd be a "release promotion" action (https://mozilla-hub.atlassian.net/browse/RELENG-981). So to trigger a release, you'd go to the Task Group page in Taskcluster of the revision you want to promote, select a `Promote Add-on` action (or similar), enter in the name of the add-on to promote, and then submit.

Reply:
> We can now beetmove TC artifacts to pub/vpn/addons
Nice! - However we still have a dependency to setup proper-release-sigining for this.

What we could do if we can push to`pub/vpn/candidates/addons`.
The current `push-addon`- already does test-sign the addons with a garbage-key. 
On any push to main - if we could get the output of that beetmoved to archive, ([unzipped, so we can address each file like here](https://firefox-ci-tc.services.mozilla.com/tasks/index/mozillavpn.v2.mozilla-vpn-client.branch.releases.2.10.latest.addons/stage), that would be super cool!

Reply:
I'm happy to use whatever URL paths the VPN team likes, but I was thinking it would be something like:
```
pub/vpn/addons/{addon-name}/candidates/{version}/
pub/vpn/addons/{addon-name}/releases/{version}/
```

Since the addons don't need to be signed, there's not really any harm in beetmoving ahead of time IMO (at least to the candidates dir).

Reply:
Trying out an [experiment in staging](https://github.com/mozilla-releng/staging-mozilla-vpn-client/tree/releases/gabriel/addons_beetmover) to beetmove the addons from the `push-addon` task.

[Running into this issue with the `addons-bundle` task](https://firefox-ci-tc.services.mozilla.com/tasks/K2d6XQEdRtCeVXKkVS9b7w/runs/0/logs/public/logs/live.log#L53) ❌ 

`bash: ./taskcluster/scripts/addons/build.sh: Permission denied`

I looked into it, not much luck, thought it might be something with the in-tree wasm image, but [everything in the working directory should have permissions to run](https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/taskcluster/docker/wasm/Dockerfile#L15) 😞